src/GitPrompt.ps1

```diff
@@ -128,11 +130,18 @@ if (Get-Module NuGet) {
 }
 
 function Write-Prompt($Object, $ForegroundColor, $BackgroundColor = -1) {
+    if ($GitPromptSettings.AnsiConsole) {
```

Comment:
So in the PromptRenderer class I was working on, the idea was that it maintained an internal StringBuilder (for ANSI hosts) and it uses Write-Host for non-ANSI and the StringBuilder for ANSI.  Currently it takes an array of TextSpan objs (text, fg, bg in a class) but I was thinking it might be better to have a BeginWrite/Write/EndWrite set of methods that each individually take a TextSpan object.